src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.FreeBSD.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    // FreeBSD implementation of FileSystemWatcher's PAL.  The main partial definition of FileSystemWatcher expects:
+    // - StartRaisingEvents
+    // - StopRaisingEvents
+    // - FinalizeDispose
+
+    // We will need to implement these methods to support FreeBSD.
+    // Until then, this is stubbed out with PlatformNotSupportedException.
+
+    public partial class FileSystemWatcher
+    {
+        private void StartRaisingEvents()
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
If I understand correctly FreeBSD support will be implemented later so `NotImplementedException` should be used here